src/app/user/actions.ts

```diff
@@ -0,0 +1,26 @@
+"use server"
+
+import { getCurrentUser } from "@/app/cookies";
+import { User } from "@/db/auth";
+import { getPropertiesByUser } from "@/db/homeseeker/property";
+import { getUpcomingAppointmentsByBroker, getUpcomingAppointmentsByProperty } from "@/db/homeseeker/appointment";
+
+export async function getUserProperties(user_id: number) {
+    const properties = await getPropertiesByUser(user_id);
+    return properties;
+}
+
+export async function getPropertyAppointments(user_id: number) {
+    const appointments = await getUpcomingAppointmentsByProperty(user_id);
+    return appointments;
+}
+
+export async function getUserAppointments(user_id: number) {
+    const appointments = await getUpcomingAppointmentsByBroker(user_id);
+    return appointments;
+}
+
+export async function fetchUserDetails(): Promise<User | null> {
+    const user = await getCurrentUser();
+    return user;
+}
```

Comment:
With all of these simple actions you should be able to just return the promise directly from the database function. For example,
```ts
return getPropertiesByUser(user_id);
```

Reply:
I think it was like this so when developing i can console the output and see if it is correct. But yea once everything is done correctly, these can be simplified.